synapse/config/repository.py

```diff
@@ -180,12 +186,17 @@ def read_config(self, config, **kwargs):
                     e.message  # noqa: B306, DependencyException.message is a property
                 )
 
+            proxies = ["HTTP_PROXY", "HTTPS_PROXY"]
             if "url_preview_ip_range_blacklist" not in config:
-                raise ConfigError(
-                    "For security, you must specify an explicit target IP address "
-                    "blacklist in url_preview_ip_range_blacklist for url previewing "
-                    "to work"
-                )
+                if not any(proxy in os.environ for proxy in proxies):
```

Comment:
this will suppress the warning if *either* `HTTP_PROXY` *or* `HTTPS_PROXY` is set.

I think we should show the warning unless *both* are set.

Reply:
Just to be clear, this doesn't have to do with the ConfigError correct? Or did you want me to change both proxy conditional statements so it raises the error only if HTTP_PROXY AND HTTPS_PROXY are both set as well as have the logger warning display when HTTP_PROXY AND HTTPS_PROXY are both set.
So something like:
```
if "url_preview_ip_range_blacklist" not in config:
    if all(proxy not in proxy_env for proxy in proxies):
        raise ConfigError(...)
else:
    if all(proxy in proxy_env for proxy in proxies):
        logger.warning("".join(HTTP_PROXY_SET_WARNING))
```
What would the expected behavior be if either is set? Do we want to raise another error mentioning that both need to be set if they'd like to use the proxy option?

Reply:
I don't quite follow your question, but my point is this: if `HTTP_PROXY` is set, and `HTTPS_PROXY` is not set, then Synapse will make requests for `https` urls itself. So, the user should set a blacklist for security, and should see a warning if they do not set it.

Reply:
As an aside: I think the `if not any` syntax is probably overcomplicated when you only have two things to worry about. Why not just write:

```
if 'http' not in proxies or 'https' not in proxies
```

It's shorter and easier to understand, imho.

Reply:
Sounds good. I'll change it to just be a boolean statement for http/https then. I wasn't sure if it would be better to have it in a list incase  list does end up growing in the future to accommodate for other proxies but I guess the change is easy enough that we'll cross that bridge if it ever were to come up.

Also just to clarify you mean a boolean AND correct not a boolean OR since you mentioned showing the warning unless both are set.

> this will suppress the warning if _either_ `HTTP_PROXY` _or_ `HTTPS_PROXY` is set.
> 
> I think we should show the warning unless _both_ are set.

Reply:
I'm pretty sure I mean a boolean OR.

We should show the warning if *any* of the following are true:
 * `HTTP_PROXY` is not set
 * `HTTPS_PROXY` is not set
 * neither is set

Do you disagree?

Reply:
No I agree. Just wanted to make sure because that was my original intent with the `any` syntax.